plugins/src/android/OptimoveSDKPlugin.java

```diff
@@ -138,4 +320,142 @@ private void getCurrentUserIdentifier(CallbackContext callbackContext) {
+        try {
+            Optimove.getInstance().pushRegister();
+            callbackContext.success();
+        } catch (Exception e) {
+            callbackContext.error(e.getMessage());
+            e.printStackTrace();
+        }
+
+    }
+
+    private void inAppUpdateConsent(JSONArray args, CallbackContext callbackContext) {
+        final boolean consented;
+
+        try {
+            consented = args.getBoolean(0);
+            OptimoveInApp.getInstance().updateConsentForUser(consented);
+            callbackContext.success();
+        } catch (Exception e) {
+            e.printStackTrace();
+            callbackContext.error(e.getMessage());
+        }
+    }
+
+    private void inAppGetInboxItems(CallbackContext callbackContext) {
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US);
+        formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        List<InAppInboxItem> items = OptimoveInApp.getInstance().getInboxItems();
+        JSONArray results = new JSONArray();
+        try {
+            for (InAppInboxItem item : items) {
+                JSONObject mapped = new JSONObject();
+
+                mapped.put("id", item.getId());
+                mapped.put("title", item.getTitle());
+                mapped.put("subtitle", item.getSubtitle());
+                mapped.put("isRead", item.isRead());
+                mapped.put("sentAt", formatter.format(item.getSentAt()));
+
+                Date availableFrom = item.getAvailableFrom();
+                Date availableTo = item.getAvailableTo();
+                Date dismissedAt = item.getDismissedAt();
+                mapped.put("data", item.getData());
+
+                URL imageUrl = item.getImageUrl();
+                mapped.put("imageUrl", imageUrl == null ? null : imageUrl.toString());
+
+                if (null == availableFrom) {
+                    mapped.put("availableFrom", "");
+                } else {
+                    mapped.put("availableFrom", formatter.format(availableFrom));
+                }
+
+                if (null == availableTo) {
+                    mapped.put("availableTo", "");
+                } else {
+                    mapped.put("availableTo", formatter.format(availableTo));
+                }
+
+                if (null == dismissedAt) {
+                    mapped.put("dismissedAt", "");
+                } else {
+                    mapped.put("dismissedAt", formatter.format(dismissedAt));
+                }
+
+                results.put(mapped);
+            }
+        } catch (JSONException e) {
+            e.printStackTrace();
+            callbackContext.error(e.getMessage());
+        }
+
+        callbackContext.success(results);
+    }
+
+    private void inAppMarkAllInboxItemsAsRead(CallbackContext callbackContext) {
+        try {
+            OptimoveInApp.getInstance().markAllInboxItemsAsRead();
+            callbackContext.success();
+        } catch (Exception e) {
```

Comment:
implementation in kumulos SDK is

```
    private void inAppMarkAllInboxItemsRead(CallbackContext callbackContext) {
        boolean result = KumulosInApp.markAllInboxItemsAsRead(cordova.getContext());
        if (result) {
            callbackContext.success();
        }
        else {
            callbackContext.error("Failed to mark all messages as read");
        }
    }
```

Can you check, do we ever throw exception here? Was kumulos implementation wrong, what's the story?

Reply:
to stay consistent with `inAppMarkAsRead` send error response when result is false